### PR TITLE
build, ci: Add file-based logging for individual packages

### LINKED ISSUE #16368: depends: File-based logging for individual packages
We should have file-based logging for each depends package so that:

1. We can diff logs to spot regressions when changing our build script / updating packages.
2. Travis has a line limit for output, so when running on Travis, we could withhold logs until we notice a non-zero exit code, then print `stdout`/`stderr`.

### PR DESCRIPTION
**Pull request by hebasto**
This PR adds file-based logging for individual packages in depends. To use this feature one should provide `LOG=1`.

A log file is printed out automatically in case of a build error. After successful build log files are being moved along with package archives:
```
$ make -C depends HOST=x86_64-w64-mingw32 LOG=1
$ find ./depends/built/x86_64-w64-mingw32 -name '*.log' | sort
./depends/built/x86_64-w64-mingw32/bdb/bdb-4.8.30-5100a099801.log
./depends/built/x86_64-w64-mingw32/boost/boost-1_71_0-313f82dc7de.log
./depends/built/x86_64-w64-mingw32/libevent/libevent-2.1.12-stable-3fa27048d5e.log
./depends/built/x86_64-w64-mingw32/libnatpmp/libnatpmp-4536032ae32268a45c073a4d5e91bbab4534773a-9db4850dd32.log
./depends/built/x86_64-w64-mingw32/miniupnpc/miniupnpc-2.2.2-75d9a1807e0.log
./depends/built/x86_64-w64-mingw32/native_b2/native_b2-1_71_0-3bf253c19bf.log
./depends/built/x86_64-w64-mingw32/qrencode/qrencode-3.4.4-dfac87af599.log
./depends/built/x86_64-w64-mingw32/qt/qt-5.15.2-9304e03d3ac.log
./depends/built/x86_64-w64-mingw32/sqlite/sqlite-3320100-455acafa7be.log
./depends/built/x86_64-w64-mingw32/zeromq/zeromq-4.3.1-5ff627ec84a.log
```

An example of CI tasks with package build errors -- https://cirrus-ci.com/task/5275741788045312

Closes #16368.

### CODE REVIEW

**Comment by practicalswift** [comment]
Concept ACK

**Comment by hebasto** [comment]
cc @dongcarl

**Comment by MarcoFalke** [comment]
Concept ACK. Does the ci print the log on failure?

**Comment by hebasto** [comment]
Updated 95b5dfa4da8c876c7c1e9e0bbc8ae9ccb97c9370 -> 68459dd31ad6dced394ae6d5d4bd07fe3fd8f510 ([pr19952.01](https://github.com/hebasto/bitcoin/commits/pr19952.01) -> [pr19952.02](https://github.com/hebasto/bitcoin/commits/pr19952.02), [diff](https://github.com/hebasto/bitcoin/compare/pr19952.01..pr19952.02)):

- addressed @MarcoFalke's [comment](https://github.com/bitcoin/bitcoin/pull/19952#issuecomment-702658581):
> Concept ACK. Does the ci print the log on failure?

An example of failure logging in CI -- https://travis-ci.org/github/hebasto/bitcoin/builds/732364252 with the following [patch](https://github.com/hebasto/bitcoin/commit/2b12f283dd7ddfe39fb5751b73a61fd2b039ec75):
```diff
--- a/depends/patches/qt/freetype_back_compat.patch
+++ b/depends/patches/qt/freetype_back_compat.patch
@@ -22,7 +22,7 @@ index 3f543755..8ecc1c8c 100644
      }
  #if defined(FT_FONT_FORMATS_H)
 -    const char *fmt = FT_Get_Font_Format(face);
-+    const char *fmt = FT_Get_X11_Font_Format(face);
++    const char *fmt = FT_Get_X11_Font_Format(face)
      if (fmt && qstrncmp(fmt, "CFF", 4) == 0) {
          FT_Bool no_stem_darkening = true;
          FT_Error err = FT_Property_Get(qt_getFreetype(), "cff", "no-stem-darkening", &no_stem_darkening);
```

**Comment by hebasto** [comment]
Rebased 68459dd31ad6dced394ae6d5d4bd07fe3fd8f510 -> 4b676b1757a397ea39e6fabff42c3b30c4489630 ([pr19952.02](https://github.com/hebasto/bitcoin/commits/pr19952.02) -> [pr19952.03](https://github.com/hebasto/bitcoin/commits/pr19952.03)).

File: depends/README.md [hunk 1]
```diff
@@ -121,6 +121,8 @@ build script logic) are searched for among the host system packages using
 <dd>(EXPERTS ONLY) When cross-compiling for macOS, use Clang found in the
 system's <code>$PATH</code> rather than the default prebuilt release of Clang
 from llvm.org. Clang 8 or later is required.</dd>
+<dt>LOG</dt>
+<dd>Use file-based logging for individual packages</dd>
```
Comment (dongcarl) [inline depends/README.md:125]: Could you add more details? For example, where would we expect these log files to appear?

File: depends/funcs.mk [hunk 2]
```diff
@@ -79,7 +80,7 @@ $(1)_download_path_fixed=$(subst :,\:,$$($(1)_download_path))
 # The default behavior for tar will try to set ownership when running as uid 0 and may not succeed, --no-same-owner disables this behavior
 $(1)_fetch_cmds ?= $(call fetch_file,$(1),$(subst \:,:,$$($(1)_download_path_fixed)),$$($(1)_download_file),$($(1)_file_name),$($(1)_sha256_hash))
 $(1)_extract_cmds ?= mkdir -p $$($(1)_extract_dir) && echo "$$($(1)_sha256_hash)  $$($(1)_source)" > $$($(1)_extract_dir)/.$$($(1)_file_name).hash &&  $(build_SHA256SUM) -c $$($(1)_extract_dir)/.$$($(1)_file_name).hash && tar --no-same-owner --strip-components=1 -xf $$($(1)_source)
-$(1)_preprocess_cmds ?=
+$(1)_preprocess_cmds ?= true
```
Comment (dongcarl) [inline depends/funcs.mk:83]: Why is this necessary?

File: depends/funcs.mk [hunk 3]
```diff
@@ -173,6 +174,10 @@ endif
 endef
 
 define int_add_cmds
+ifneq ($(LOG),)
+$(1)_logging = >>$$($(1)_build_log) 2>&1 || { if test -f $$($(1)_build_log); then cat $$($(1)_build_log); fi; exit 1; }
```
Comment (dongcarl) [inline depends/funcs.mk:178]: So this would mean that while the `$(1)_*_cmds` is running, we wouldn't see any logs, and only if the `cmds` fail, would we `cat` the logs? Am I understanding this correctly?

Comment (hebasto) [reply depends/funcs.mk:178]: Yes, you're correct. This behavior is applied only if when `LOG` is set.

File: depends/funcs.mk (hunk 2 above)
Comment (hebasto) [reply depends/funcs.mk:83]: To apply `$(1)_logging` to `$(1)_preprocess_cmds` even the latter is not set explicitly.

File: depends/README.md (hunk 1 above)
Comment (hebasto) [reply depends/README.md:125]: If wording in the updated OP is ok, I'll push it into the docs.

**Comment by laanwj** [comment]
Concept ACK, I think not being verbose unless the level of detail is really needed (say, on a failure) is good for build/test commands. Makes it easier to spot actual problems.
Please do make it easy to figure out where the logs went in case they're needed though :slightly_smiling_face:

**Comment by hebasto** [comment]
Updated 4b676b1757a397ea39e6fabff42c3b30c4489630 -> 46e2ff7c0dd4d15d17dc6d9015f28191d3efc05b ([pr19952.03](https://github.com/hebasto/bitcoin/commits/pr19952.03) -> [pr19952.04](https://github.com/hebasto/bitcoin/commits/pr19952.04)):

- rebased on top of the recent CI changes
- addressed @laanwj's [comment](https://github.com/bitcoin/bitcoin/pull/19952#issuecomment-730834978):

> Please do make it easy to figure out where the logs went in case they're needed though

The example of CI output in case of error during depends build is [here](https://cirrus-ci.com/build/4583635837779968).

**Comment by dongcarl** [comment]
Hmmm, I've been thinking about this... And wondering if we could use `tee` and process substitute to do:

```diff
diff --git a/depends/funcs.mk b/depends/funcs.mk
index 3ddeaf0378..2de5f67b5c 100644
--- a/depends/funcs.mk
+++ b/depends/funcs.mk
@@ -80,7 +80,7 @@ $(1)_download_path_fixed=$(subst :,\:,$$($(1)_download_path))
 # The default behavior for tar will try to set ownership when running as uid 0 and may not succeed, --no-same-owner disables this behavior
 $(1)_fetch_cmds ?= $(call fetch_file,$(1),$(subst \:,:,$$($(1)_download_path_fixed)),$$($(1)_download_file),$($(1)_file_name),$($(1)_sha256_hash))
 $(1)_extract_cmds ?= mkdir -p $$($(1)_extract_dir) && echo "$$($(1)_sha256_hash)  $$($(1)_source)" > $$($(1)_extract_dir)/.$$($(1)_file_name).hash &&  $(build_SHA256SUM) -c $$($(1)_extract_dir)/.$$($(1)_file_name).hash && tar --no-same-owner --strip-components=1 -xf $$($(1)_source)
-$(1)_preprocess_cmds ?= true
+$(1)_preprocess_cmds ?=
 $(1)_build_cmds ?=
 $(1)_config_cmds ?=
 $(1)_stage_cmds ?=
@@ -177,7 +177,7 @@ endef
 
 define int_add_cmds
 ifneq ($(LOG),)
-$(1)_logging = >>$$($(1)_build_log) 2>&1 || { if test -f $$($(1)_build_log); then cat $$($(1)_build_log); fi; exit 1; }
+$(1)_logging = exec >> >(tee -a $$($(1)_build_log)) 2>&1;
 endif
 
 $($(1)_fetched):
@@ -196,23 +196,23 @@ $($(1)_preprocessed): | $($(1)_extracted)
 	$(AT)echo Preprocessing $(1)...
 	$(AT)mkdir -p $$(@D) $($(1)_patch_dir)
 	$(AT)$(foreach patch,$($(1)_patches),cd $(PATCHES_PATH)/$(1); cp $(patch) $($(1)_patch_dir) ;)
-	$(AT){ cd $$(@D); $(call $(1)_preprocess_cmds, $(1)); } $$($(1)_logging)
+	$(AT)( $$($(1)_logging) cd $$(@D); $(call $(1)_preprocess_cmds, $(1)) )
 	$(AT)touch $$@
 $($(1)_configured): | $($(1)_dependencies) $($(1)_preprocessed)
 	$(AT)echo Configuring $(1)...
 	$(AT)rm -rf $(host_prefix); mkdir -p $(host_prefix)/lib; cd $(host_prefix); $(foreach package,$($(1)_all_dependencies), tar --no-same-owner -xf $($(package)_cached); )
 	$(AT)mkdir -p $$(@D)
-	$(AT)+{ cd $$(@D); $($(1)_config_env) $(call $(1)_config_cmds, $(1)); } $$($(1)_logging)
+	$(AT)+( $$($(1)_logging) cd $$(@D); $($(1)_config_env) $(call $(1)_config_cmds, $(1)) )
 	$(AT)touch $$@
 $($(1)_built): | $($(1)_configured)
 	$(AT)echo Building $(1)...
 	$(AT)mkdir -p $$(@D)
-	$(AT)+{ cd $$(@D); $($(1)_build_env) $(call $(1)_build_cmds, $(1)); } $$($(1)_logging)
+	$(AT)+( $$($(1)_logging) cd $$(@D); $($(1)_build_env) $(call $(1)_build_cmds, $(1)) )
 	$(AT)touch $$@
 $($(1)_staged): | $($(1)_built)
 	$(AT)echo Staging $(1)...
 	$(AT)mkdir -p $($(1)_staging_dir)/$(host_prefix)
-	$(AT)+{ cd $($(1)_build_dir); $($(1)_stage_env) $(call $(1)_stage_cmds, $(1)); } $$($(1)_logging)
+	$(AT)+( $$($(1)_logging) cd $($(1)_build_dir); $($(1)_stage_env) $(call $(1)_stage_cmds, $(1)) )
 	$(AT)rm -rf $($(1)_extract_dir)
 	$(AT)touch $$@
 $($(1)_postprocessed): | $($(1)_staged)
```

I supposed POSIX sh doesn't have process substitution... But perhaps for depends we can set `.SHELL = bash`? Just thinking...

**Comment by hebasto** [comment]
> I supposed POSIX sh doesn't have process substitution...

https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#exec

**Comment by hebasto** [comment]
Updated 46e2ff7c0dd4d15d17dc6d9015f28191d3efc05b -> bdd1056fbdd2164f852940afbaf63abe83d1f571 ([pr19952.04](https://github.com/hebasto/bitcoin/commits/pr19952.04) -> [pr19952.05](https://github.com/hebasto/bitcoin/commits/pr19952.05)) due to the conflict with #22283.

**Comment by hebasto** [comment]
The PR description has been updated.

**Comment by hebasto** [comment]
Rebased bdd1056fbdd2164f852940afbaf63abe83d1f571 -> 0b9357beb4bb6e39125090204b26a00842f2736e ([pr19952.05](https://github.com/hebasto/bitcoin/commits/pr19952.05) -> [pr19952.06](https://github.com/hebasto/bitcoin/commits/pr19952.06)) due to the conflict with #24212.

**Comment by hebasto** [comment]
Rebased 0b9357beb4bb6e39125090204b26a00842f2736e -> 3c5c5839fe611e5bb723c792a24c582df4b0f5c0 ([pr19952.06](https://github.com/hebasto/bitcoin/commits/pr19952.06) -> [pr19952.07](https://github.com/hebasto/bitcoin/commits/pr19952.07)) due to the conflict with #23998.

**Comment by hebasto** [comment]
#### Guix builds:
```
$ find guix-build-$(git rev-parse --short=12 HEAD)/output/ -type f -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum
e0e8d12c43c077aa59be7ef51d0290c1c0a204592b03a024d7e98054547b4298  guix-build-3c5c5839fe61/output/aarch64-linux-gnu/SHA256SUMS.part
2b04b0c30974d5ec8bb048890a4c89e31afb60b7e638155b73c96a0cb07603a6  guix-build-3c5c5839fe61/output/aarch64-linux-gnu/bitcoin-3c5c5839fe61-aarch64-linux-gnu-debug.tar.gz
cd8be3cdd7a54877dce307072b646e3706b57f626e2ea13f38274394aa9ae16f  guix-build-3c5c5839fe61/output/aarch64-linux-gnu/bitcoin-3c5c5839fe61-aarch64-linux-gnu.tar.gz
9a760fb5b0b1f7a7c1d35cac6b4447eea77823722230d3bf19d8a809ac7affdc  guix-build-3c5c5839fe61/output/arm-linux-gnueabihf/SHA256SUMS.part
fc53bb9ec64fa928baf787119174ba2dc8a48f982eec85554073a04cf5a8416f  guix-build-3c5c5839fe61/output/arm-linux-gnueabihf/bitcoin-3c5c5839fe61-arm-linux-gnueabihf-debug.tar.gz
c7720367ef22d2f2f622cfdee492d7101eae3633c7b61ee9035837bd267438c0  guix-build-3c5c5839fe61/output/arm-linux-gnueabihf/bitcoin-3c5c5839fe61-arm-linux-gnueabihf.tar.gz
e064e261db13ebeafcb9bcc1ca8b1fca16d8105a9e4454b6f240cd055236789b  guix-build-3c5c5839fe61/output/arm64-apple-darwin/SHA256SUMS.part
3bb7a65fb83ebf0b2d06c2d1d1e9bee096198a0c7f9f7b8c41e0f7d9a4f85650  guix-build-3c5c5839fe61/output/arm64-apple-darwin/bitcoin-3c5c5839fe61-arm64-apple-darwin.tar.gz
39b9d88cb56e9a2e5626c7673f6bcc2764ac341a9bd9d4cf048900543261013c  guix-build-3c5c5839fe61/output/arm64-apple-darwin/bitcoin-3c5c5839fe61-osx-unsigned.dmg
8d5b5acd83883c53228527ca87be6af46c765243d8bdab073ec2355c110394dd  guix-build-3c5c5839fe61/output/arm64-apple-darwin/bitcoin-3c5c5839fe61-osx-unsigned.tar.gz
2f605b0320c90cd30028724d3684971c0b323135d5bdc3894e826b54aab5a79d  guix-build-3c5c5839fe61/output/dist-archive/bitcoin-3c5c5839fe61.tar.gz
e4d5faaa07788ed1b454ecbbcdc81d1546239a8356fa3c118dc4d4714ad2e771  guix-build-3c5c5839fe61/output/powerpc64-linux-gnu/SHA256SUMS.part
3a2401d8b989b7f9a714f27d4c8894040d5407e0d4d841adc705ff420a119467  guix-build-3c5c5839fe61/output/powerpc64-linux-gnu/bitcoin-3c5c5839fe61-powerpc64-linux-gnu-debug.tar.gz
f86901a9f2336201f5a3cac2556ad317e0ef33a1a89303557ee0ceb5ee31c43d  guix-build-3c5c5839fe61/output/powerpc64-linux-gnu/bitcoin-3c5c5839fe61-powerpc64-linux-gnu.tar.gz
b7302a12063fc85a96077f9048451d737ae3a5f950ad17fc521a63d02063b6b7  guix-build-3c5c5839fe61/output/powerpc64le-linux-gnu/SHA256SUMS.part
d181f8ee35239094d3dfa953bd8ba3d51a8ee38a4038149a15f400d386587274  guix-build-3c5c5839fe61/output/powerpc64le-linux-gnu/bitcoin-3c5c5839fe61-powerpc64le-linux-gnu-debug.tar.gz
810d60570ca083217fc2eb378bceb5361605dda47ae2fd76bc693a027ee8409a  guix-build-3c5c5839fe61/output/powerpc64le-linux-gnu/bitcoin-3c5c5839fe61-powerpc64le-linux-gnu.tar.gz
b2f8d87ae3d8ce4abb121b7f589f9d1db71ee32a456fec3639e7714d0fa7ca5f  guix-build-3c5c5839fe61/output/riscv64-linux-gnu/SHA256SUMS.part
994d00a6514c704ac177f48790b9acb586f1e56f8aa702b205d532d95a81e2d9  guix-build-3c5c5839fe61/output/riscv64-linux-gnu/bitcoin-3c5c5839fe61-riscv64-linux-gnu-debug.tar.gz
b4bacd388be1f687404baa31650647e3416b743d0a8ceb4a57121d25750cb0fc  guix-build-3c5c5839fe61/output/riscv64-linux-gnu/bitcoin-3c5c5839fe61-riscv64-linux-gnu.tar.gz
f285e4da8535f25605f3f16e3f72b8a8f64210ec4686b5aba784ff48842476d9  guix-build-3c5c5839fe61/output/x86_64-apple-darwin/SHA256SUMS.part
af453058f80b0a933563ab5c3f8003af75e37d79fff823f418ba881c9e53ed81  guix-build-3c5c5839fe61/output/x86_64-apple-darwin/bitcoin-3c5c5839fe61-osx-unsigned.dmg
7eb22cd46617f0c1a55328931e3fa5e7a9bb481ff56dc136fe62854ef08dde99  guix-build-3c5c5839fe61/output/x86_64-apple-darwin/bitcoin-3c5c5839fe61-osx-unsigned.tar.gz
3456ac4af66f2e71473a3d891c9e953a39a49cbd306d8d9d935e50aa50dea8d8  guix-build-3c5c5839fe61/output/x86_64-apple-darwin/bitcoin-3c5c5839fe61-osx64.tar.gz
d35fde85151a30f780bf69c12e40399ac96810e1cb6dfa7861fb7a48d0be801d  guix-build-3c5c5839fe61/output/x86_64-linux-gnu/SHA256SUMS.part
b1994d47c85bfa248b0483bc910d805718c2c920571e9e465d5cbe6da3cae65b  guix-build-3c5c5839fe61/output/x86_64-linux-gnu/bitcoin-3c5c5839fe61-x86_64-linux-gnu-debug.tar.gz
591e0d6d7a706d10b5bcbe93387560b6790dd989f046d0a32ec20e028bbda037  guix-build-3c5c5839fe61/output/x86_64-linux-gnu/bitcoin-3c5c5839fe61-x86_64-linux-gnu.tar.gz
```

**Comment by hebasto** [comment]
Rebased 3c5c5839fe611e5bb723c792a24c582df4b0f5c0 -> eb4b01edf0a4cbe3e7b306f1cb51f03f87f13879 ([pr19952.07](https://github.com/hebasto/bitcoin/commits/pr19952.07) -> [pr19952.08](https://github.com/hebasto/bitcoin/commits/pr19952.08)) due to the conflict with #24522.

**Comment by hebasto** [comment]
Friendly ping @dongcarl @MarcoFalke

**Comment by MarcoFalke** [comment]
Concept ACK

**Comment by hebasto** [comment]
Rebased eb4b01edf0a4cbe3e7b306f1cb51f03f87f13879 -> 86c2889518df0234b8b28112a9771b85c577eefd ([pr19952.08](https://github.com/hebasto/bitcoin/commits/pr19952.08) -> [pr19952.09](https://github.com/hebasto/bitcoin/commits/pr19952.09)) due to the conflict with #24285.

**Comment by laanwj** [comment]
Going to test this.

**Comment by laanwj** [comment]
Tested ACK 86c2889518df0234b8b28112a9771b85c577eefd

I know it's not the intended use but I patched this into the guix build, as that's the only depends build i do at the moment:
```patch
--- a/contrib/guix/libexec/build.sh
+++ b/contrib/guix/libexec/build.sh
@@ -198,7 +198,7 @@ esac
 ####################
 
 # Build the depends tree, overriding variables that assume multilib gcc
-make -C depends --jobs="$JOBS" HOST="$HOST" \
+make -C depends --jobs="$JOBS" HOST="$HOST" LOG=1 \
                                    ${V:+V=1} \
                                    ${SOURCES_PATH+SOURCES_PATH="$SOURCES_PATH"} \
                                    ${BASE_CACHE+BASE_CACHE="$BASE_CACHE"} \
```

The log was clean:
```
make: Entering directory '/bitcoin/depends'
Extracting boost...
/home/guest/sources/boost_1_77_0.tar.bz2: OK
Preprocessing boost...
Configuring boost...
Building boost...
Staging boost...
Postprocessing boost...
Caching boost...
Extracting libevent...
/home/guest/sources/libevent-2.1.12-stable.tar.gz: OK
Preprocessing libevent...
Configuring libevent...
⋮ 
Preprocessing zeromq...
Configuring zeromq...
Building zeromq...
Staging zeromq...
Postprocessing zeromq...
Caching zeromq...
copying packages: boost libevent qt expat libxcb xcb_proto libXau xproto freetype fontconfig libxkbcommon libxcb_util libxcb_util_render libxcb_util_keysyms libxcb_util_image libxcb_util_wm qrencode bdb sqlite miniupnpc libnatpmp systemtap zeromq
to: /bitcoin/depends/riscv64-linux-gnu
```

After the build, the individual log files could be found in the cache:
```
./base-cache/riscv64-linux-gnu/qt/qt-5.15.3-30a361c8788.log
./base-cache/riscv64-linux-gnu/libnatpmp/libnatpmp-4536032ae32268a45c073a4d5e91bbab4534773a-59aa9cc4671.log
./base-cache/riscv64-linux-gnu/qrencode/qrencode-3.4.4-0469a7d34c9.log
./base-cache/riscv64-linux-gnu/boost/boost-1.77.0-1f1d64dbf3c.log
./base-cache/riscv64-linux-gnu/libxcb_util_render/libxcb_util_render-0.3.9-95703f260b3.log
⋮ 
./base-cache/riscv64-linux-gnu/bdb/bdb-4.8.30-cf38a657499.log
./base-cache/riscv64-linux-gnu/sqlite/sqlite-3320100-06ea4df8e5e.log
./base-cache/riscv64-linux-gnu/libxcb_util_keysyms/libxcb_util_keysyms-0.4.0-baf1d726a20.log
./base-cache/riscv64-linux-gnu/fontconfig/fontconfig-2.12.6-181950ffe12.log
./base-cache/riscv64-linux-gnu/expat/expat-2.4.1-24df0173864.log
```

I also corrupted a patch intentionally to see if the logging would still report the error in that case:
```
make: Entering directory '/bitcoin/depends'
Extracting qt...
/home/guest/sources/qtbase-everywhere-opensource-src-5.15.3.tar.xz: OK
/home/guest/sources/qttranslations-everywhere-opensource-src-5.15.3.tar.xz: OK
/home/guest/sources/qttools-everywhere-opensource-src-5.15.3.tar.xz: OK
Preprocessing qt...
patching file qtbase/configure
patch: **** malformed patch at line 19: dsoiusaoifasdfiadfiosudfsduf

make: *** [funcs.mk:289: /bitcoin/depends/work/build/riscv64-linux-gnu/qt/5.15.3-3a31e8c8789/.stamp_preprocessed] Error 1
make: Leaving directory '/bitcoin/depends'
```
